### PR TITLE
add config option for stripping comain off web server auth

### LINKED ISSUE #223: Web Server Auth strips domain from login
Hi,

We have users with different domains in out LDAP, so our UIDs are complete email addresses.
With "Use Web Server Auth", the domain part is stripped off the REMOTE_USER before checking if the user exists/synchronizing.
This was a bit of a surprise.

Changing the line to
`$this->login = $webServerAuthUser;`
made it work for me.

I think the removal of the domain should be configurable and/or be documented.

https://github.com/matomo-org/plugin-LoginLdap/blob/eae0cb494dc424fb2d710db7c5170822cad32bff/Auth/WebServerAuth.php#L68

Regards,
  EmTeedee

### PR DESCRIPTION
**Pull request by EmTeedee**
This is my attempt at fixing this, it works in our installation.
Default is to keep previous behavior.

Fixes matomo-org/plugin-LoginLdap#223

### CODE REVIEW

File: Config.php [hunk 1]
```diff
@@ -186,6 +187,11 @@ public static function shouldAppendUserEmailSuffixToUsername()
         return self::getConfigOption('append_user_email_suffix_to_username') == 1;
     }
 
+    public static function getStripDomainFromWebAuth()
+    {
+        return self::getConfigOption('strip_domain_from_web_auth');
```
Comment (diosmosis) [inline Config.php]: Can we model this like the other functions that return bools? Ie, `getConfigOption(...) == 1`?

**Review by diosmosis — CHANGES_REQUESTED** [review]
Hi @EmTeedee, thanks for the great contribution! I requested one small change, but otherwise this PR looks good. Once it's made I'll merge it.

File: Config.php (hunk 1 above)
Comment (EmTeedee) [reply Config.php]: Sure thing, 'twas but an oversight.

**Comment by diosmosis** [comment]
@EmTeedee thanks for making the change, I've merged this! I'll release a new version sometime later today.

**Pull request merged**